components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/dao/impl/FeatureDAOImpl.java

```diff
@@ -261,6 +261,28 @@ public boolean deleteFeaturesOfProfile(Profile profile) throws FeatureManagerDAO
         }
     }
 
+    @Override
+    public boolean deleteFeaturesOfProfile(int profileId) throws FeatureManagerDAOException {
+        Connection conn;
+        PreparedStatement stmt = null;
+
+        try {
+            conn = this.getConnection();
+            String query = "DELETE FROM DM_PROFILE_FEATURES WHERE PROFILE_ID = ?";
+            stmt = conn.prepareStatement(query);
+            stmt.setInt(1, profileId);
+            stmt.executeUpdate();
+            return true;
+
+        } catch (SQLException e) {
+            String msg = "Error occurred while deleting the feature related to a profile.";
+            log.error(msg);
```

Comment:
Fixed in commit [1].
[1] https://github.com/wso2/carbon-device-mgt/commit/cdbdc86f42b350c24e9d059ff7c23e1f09242df4